db/compaction_iterator.h

```diff
@@ -21,6 +21,21 @@
 
 namespace rocksdb {
 
+class SnapshotListFetchCallback {
+ public:
+  SnapshotListFetchCallback()
+      : snap_refresh_nanos_(500 * 1000 * 1000) {} // 0.5s
+  SnapshotListFetchCallback(uint64_t snap_refresh_nanos)
+      : snap_refresh_nanos_(snap_refresh_nanos) {}
+  virtual void Refresh(std::vector<SequenceNumber>* snapshots,
+                       SequenceNumber max) = 0;
+  uint64_t snap_refresh_nanos() { return snap_refresh_nanos_; }
```

Comment:
It's not a big deal but why does snap_refresh_nanos_ belong to SnapshotListFetchCallback?

Reply:
One benefit would be less variables in compaction_iterator. But let me know if you prefer any other place.